### PR TITLE
Profile endpoints

### PR DESCRIPTION
**Pull request by DarkendHall**
Also update register error checking and adds display name to register form

### CODE REVIEW

File: frontend/src/components/PageRegister.vue [hunk 1]
```diff
@@ -86,26 +98,36 @@ function submitIfValid() {
...
-                    awaitingResponse.value = false;
-                    isSubmitted.value = false;
-                    checkErrors();
-                    console.log(data);
+                .then((response) => {
+                    if (response.ok) {
+                        response.json().then(() => {
+                            awaitingResponse.value = false;
+                            isSubmitted.value = false;
+                            router.push('/login');
```
Comment (ToniKaru) [inline frontend/src/components/PageRegister.vue:112]: Add popup for with message like: "Thank you for registering! Please login."

**Review by ToniKaru — CHANGES_REQUESTED** [review]
1. Profile frontend should display current displayName, not default "DarkendHall" anymore.
2. Should displayname be changeable? 
 - if yes, displayname needs to be updated to function so on save. 
 - if no, user should not be able to change it

**Comment by DarkendHall** [comment]
> 

This pull request doesn't connect the frontend and backend, the only changes in there is reformatting

**Comment by DarkendHall** [comment]
> 1. Profile frontend should display current displayName, not default "DarkendHall" anymore.
> 2. Should displayname be changeable?
> 
> * if yes, displayname needs to be updated to function so on save.
> * if no, user should not be able to change it

**Comment by ToniKaru** [comment]
Sounds right and trello already has a card for it. 👍

**Review by Yuriks1 — APPROVED** [review]
Well done!

**Review by ToniKaru — APPROVED** [review]
Great work!  VerifyToken() is very useful

**Pull request merged**